libraries/cbor/src/values.rs

```diff
@@ -206,7 +231,9 @@ where
     }
 }
 
+/// Trait that indicates that a type can be (fallibly) converted to a CBOR [`Value`].
```

Comment:
This is not how we actually use it. Maybe we have to rename it? What we do is implement it for `Option<T>` and just `map` back to `Option`. You could implement it for a type that fallibly converts, but that might make it less intuitive. I.e. you use `cbor_map_options!` and some keys just disappear instead of failing to compile, because they were a fallibly conversion.

Reply:
Ah, I just based the text on the method signature without seeing the use in `macros.rs` and it seems I misunderstood.

Given that this (and the `IntoCborValue` trait) are just used internally at the moment, how about making them `pub(crate)` so they don't need documenting?

Reply:
The problem is that they are used in macros, so their visibility must leak. There's usually 2 cumulable solutions for this problem:
- Add "internal" in the name of things leaking through macros (cost: depends on usage)
- Add `#[doc(hidden)]` to things leaking through macros (cost: local)

Reply:
Sounds good, I double checked: Other things I noticed in `values.rs` you can choose to make pub(crate) if you want:
- `Constants` 
- `type_label`

Reply:
> The problem is that they are used in macros, so their visibility must leak. 

Ah yes of course &ndash; I've dropped the commit that tried that.

Reply:
For the time being I've gone for what seemed the least invasive option, which is to change the doc text for those traits to be blander/more innocuous.

Reply:
Sounds good, we can don't have to bloat this PR anymore.